cidc_api/models/models.py

```diff
@@ -432,6 +437,10 @@ def insert(self, session: Session, commit: bool = True, compute_etag: bool = Tru
         # Always commit, because we don't want to grant IAM download unless this insert succeeds.
         super().insert(session=session, commit=True, compute_etag=compute_etag)
```

Comment:
does this want to be after the return of network viewers?

Reply:
No - this code inserts the permission record into the database, which is what gives a network viewer permission to view metadata for files pertaining to a certain trial/upload type (but not to download them).